src/main/groovy/com/vanniktech/maven/publish/MavenPublishPlugin.groovy

```diff
@@ -39,10 +39,27 @@ class MavenPublishPlugin extends BaseMavenPublishPlugin {
       options.linksOffline "https://developer.android.com/reference", "${project.android.sdkDirectory}/docs/reference"
     }
 
-    project.tasks.create("androidJavadocsJar", Jar.class) {
+    def androidJavadocsJar = project.tasks.create("androidJavadocsJar", Jar.class) {
       classifier = 'javadoc'
-      from project.androidJavadocs.destinationDir
-    }.dependsOn("androidJavadocs")
+    }
+
+    if (plugins.hasPlugin('kotlin-android')) {
+      def dokkaOutput = "${project.docsDir}/dokka"
+      project.plugins.apply('org.jetbrains.dokka-android')
+      project.dokka {
```

Comment:
I don't think this is quite right, and as a result I don't think Android SDK refs get linked

See:
https://github.com/Kotlin/dokka/issues/169
https://github.com/Arasthel/SpannedGridLayoutManager/blob/master/spannedgridlayoutmanager/build.gradle#L5

I don't think the default `dokka` task will be of an android type, or at least I'm not seeing these link

Reply:
Could very well be the case. I've used the latest snapshot for publishing one library of mine but there were no Android related things in there. Mind submitting a PR?

Reply:
@ZacSweers, dokka-android plugin [creates DokkaAndroidTask](https://github.com/Kotlin/dokka/blob/master/runners/android-gradle-plugin/src/main/kotlin/mainAndroid.kt#L12) with `dokka` name when it's applied.

 The guy from the issue Kotlin/dokka#169 was using `DokkaTask` instead of `DokkaAndroidTask`.


Android SDK are not added because its missing from it's missing from configuration as [it was done for javadoc](https://github.com/vanniktech/gradle-maven-publish-plugin/pull/37/files#diff-406fe0b4199eaa97bca99d8d9ffd96d0R39). Android SDK can be linked using [externalDocumentationLink](https://github.com/Kotlin/dokka/blob/c189c8c338c219b8d773666475f5ae2818461b0a/README.md#L118) property.

I will submit a PR tomorrow with the apropriare changes if none will do it before.